processing/src/main/java/io/druid/query/PrioritizedExecutorService.java

```diff
@@ -185,89 +198,138 @@ public int getQueueSize()
   {
     return delegateQueue.size();
   }
+}
 
-
-  public static class PrioritizedListenableFutureTask<V> implements RunnableFuture<V>,
-      ListenableFuture<V>,
-      PrioritizedRunnable,
-      Comparable<PrioritizedListenableFutureTask>
+class PrioritizedListenableFutureTask<V> implements RunnableFuture<V>,
+    ListenableFuture<V>,
+    PrioritizedRunnable,
+    Comparable<PrioritizedListenableFutureTask>
+{
+  // NOTE: For these LOWER means higher priority. As such we swap left and right in the compares
+  private static final Comparator<PrioritizedListenableFutureTask> PRIORITY_COMPARATOR = new Ordering<PrioritizedListenableFutureTask>()
```

Comment:
Is there a reason we need a separate comparator for something that is already comparable?